test/uk/gov/hmrc/fileupload/controllers/transfer/TransferControllerSpec.scala

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright 2016 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.fileupload.controllers.transfer
+
+import cats.data.Xor
+import org.scalatest.concurrent.ScalaFutures
+import play.api.http.Status
+import play.api.test.FakeRequest
+import uk.gov.hmrc.fileupload.transfer.Service._
+import uk.gov.hmrc.fileupload.{EnvelopeId, Support}
+import uk.gov.hmrc.play.test.{UnitSpec, WithFakeApplication}
+
+import scala.concurrent.{ExecutionContext, Future}
+
+class TransferControllerSpec extends UnitSpec with WithFakeApplication with ScalaFutures {
+
+  implicit val ec = ExecutionContext.global
+
+  val failed = Future.failed(new Exception("not good"))
+
+  def newController(softDelete: EnvelopeId => Future[SoftDeleteResult] = _ => failed) =
+    new TransferController(softDelete)
+
+
+  "Delete envelope" should {
+    "return response with 200" in {
+      val envelope = Support.envelope
+      val request = FakeRequest("DELETE", s"/envelopes/${envelope._id}")
```

Comment:
I'd keep it as FakeRequest() only as adding "DELETE" and url suggests that controllers needs it but it doesn't. These information is used by the router but here we bypass routing and pass a request directly to the controller.